### PR TITLE
Begin setting up torsion automation

### PR DESCRIPTION
**Pull request by mattwthompson**
## Development checklist
- [x] Update YAMMBS to use more than 24 cores at once when running torsion optimizations (might be useful to use fewer CPU cores until then)
- [ ] Add plotting script - do next week
- [ ] Other metrics? - do in the future

## Submission Checklist
- [ ] Created a new directory in the `submissions` directory containing the YDS input file and optionally a force field `.offxml` file
- [ ] Triggered the benchmark workflow with a PR comment of the form `/run-benchmark path/to/submission`
- [ ] Waited for the workflow to finish and a comment with `Job status: success` to be posted
- [ ] Reviewed the results committed by the workflow
- [ ] Published the corresponding Zenodo entry and retrieved the DOI
- [ ] Added the Zenodo DOI to the table in the main README
- [ ] Ready to merge!

### CODE REVIEW

File: .github/workflows/torsions.yaml [hunk 1]
```diff
@@ -145,8 +146,9 @@ jobs:
 
           input_file=${{ inputs.path }}                        # path to the input YAML file
           input_dir=$(dirname $input_file)                     # parent directory of input YAML file
-          git add $input_dir/output/{log_sse}.csv
-          git add $input_dir/output/{log_sse}.csv
+          git add $input_dir/output/log_sse.csv
```
Comment (mattwthompson) [inline .github/workflows/torsions.yaml:149]:
```suggestion
          git add $input_dir/output/rmsd.csv
          git add $input_dir/output/een.csv
```

File: .github/workflows/torsions.yaml [hunk 2]
```diff
@@ -155,14 +157,15 @@ jobs:
...
         shell: bash -l {0}
         run: |
           # TODO: make and bundle plot(s)
-          bzip2 tmp.sqlite
+          bzip2 torsions-dev.sqlite
           micromamba env export > env.yaml
           input_files=$(python get_files.py ${{ inputs.path }})
           python zenodo_upload.py --title "YDS Upload ${{ inputs.pr_number }}" \
-            tmp.sqlite.bz2 $input_dir/output/{log_sse}.csv \
+            torsions-dev.sqlite.bz2 $input_dir/output/log_sse.csv \
```
Comment (mattwthompson) [inline .github/workflows/torsions.yaml:168]:
```suggestion
            torsions-dev.sqlite.bz2 $input_dir/output/rmsd.csv \
            torsions-dev.sqlite.bz2 $input_dir/output/rmse.csv \
```

File: torsions.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+from yammbs.torsion.inputs import QCArchiveTorsionDataset
+
+from config import Config
+
+assert OpenEyeToolkitWrapper().is_available()
+
+
+def make_csvs(store, forcefield, out_dir):
+    print("getting log SSEs")
+    store.get_log_sse(forcefield, skip_check=True).to_csv(f"{out_dir}/log_sse.csv")
```
Comment (mattwthompson) [inline torsions.py:23]:
```suggestion
    store.get_rmsd(forcefield, skip_check=True).to_csv(f"{out_dir}/rmsd.csv")
    store.get_rmse(forcefield, skip_check=True).to_csv(f"{out_dir}/rmse.csv")
```

**Review by mattwthompson — COMMENTED** [review]
I'm not sure where it's called from, but `--log-level=INFO` to the `python torsions.py` as well

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
/run-torsions torsion-dev/larger/tmp.yaml devtools/torsions.yaml

**Comment by mattwthompson** [comment]
I don't think we took notes on this, so -

last week, @ntBre and I discovered that some data points (maybe 5-10%) were reporting both "no qm points" and "no mm points" which was a surprise because the molecules themselves were well-formed and processed into inputs. The problem was that YAMMBS looked up IDs by (mapped) SMILES alone, which works fine for molecules but doesn't work for torsion drives. Because a torsion drive can drive a given mapped SMILES at different atom quartets, the mapped SMILES alone cannot uniquely identify an ID and the lookup needs to also pass the dihedral indices through. See https://github.com/openforcefield/yammbs/pull/76/commits/b7df922deae23e43f1d92be3387ca2041f189724

**Comment by mattwthompson** [comment]
@ntbre could you give this a once-over (or a thorough review if you wish) and merge?

The only thing I'm not sure about is what the current `torsion-dev/` directory should look like

**Comment by ntBre** [comment]
Sure, I'll take a look first thing tomorrow. I actually deleted my `_tests` directory (analogous to `torsion-dev`)  when I finished testing in the earlier optimization tests, but I think it's also useful to keep it around and the current state looks fine to me.

File: devtools/env.yaml [hunk 4]
```diff
@@ -22,4 +22,4 @@ dependencies:
   - tqdm
 
   - pip:
-    - git+https://github.com/openforcefield/yammbs.git@0.1.1
+    - git+https://github.com/openforcefield/yammbs.git@add-torsion-models
```
Comment (ntBre) [inline devtools/env.yaml:25]: Do you want to update this to the latest yammbs release before merging?

File: devtools/torsions.yaml
Comment (ntBre) [inline devtools/torsions.yaml:1]: And is this env still necessary?

Comment (mattwthompson) [reply devtools/torsions.yaml:1]: Nope!

File: .github/workflows/torsions.yaml [hunk 5]
```diff
@@ -145,8 +146,10 @@ jobs:
...
           input_file=${{ inputs.path }}                        # path to the input YAML file
           input_dir=$(dirname $input_file)                     # parent directory of input YAML file
-          git add $input_dir/output/{log_sse}.csv
-          git add $input_dir/output/{log_sse}.csv
+          git add $input_dir/output/rmsd.csv
+          git add $input_dir/output/een.csv
+          git add $input_dir/output/minimized.json
+          git add $input_dir/output/metrics.json
           git commit -m "Add benchmark results"
           git push
```
[the remark this answers is not in the document]
Comment (mattwthompson) [reply .github/workflows/torsions.yaml:154]: Good call, I've opened a separate issue because I don't want to fix that right now

**Comment by mattwthompson** [comment]
Thanks!

**Pull request merged**